src/app/proposals/draft/utils/getInputData.ts

```diff
@@ -16,29 +67,184 @@ export function getInputData(
+      }[];
+
+      proposal.approval_options.forEach((option) => {
+        const formattedOption = {
+          budgetTokensSpent: BigInt(0),
+          targets: [] as `0x${string}`[],
+          values: [] as bigint[],
+          calldatas: [] as `0x${string}`[],
+          description: option.title,
+        };
+
+        option.transactions.forEach((t) => {
+          if (isTransfer(t.calldata)) {
+            const {
+              args: [_recipient, amount],
+            } = decodeFunctionData({
+              abi: transferABI,
+              data: t.calldata as `0x${string}`,
+            });
+
+            formattedOption.budgetTokensSpent += amount;
+            formattedOption.targets.push(t.target as `0x${string}`);
+            formattedOption.values.push(BigInt(0));
+            formattedOption.calldatas.push(t.calldata as `0x${string}`);
+          } else {
+            formattedOption.targets.push(
+              ethers.getAddress(t.target) as `0x${string}`
+            );
+            formattedOption.values.push(parseEther(t.value.toString() || "0"));
+            formattedOption.calldatas.push(t.calldata as `0x${string}`);
+          }
+        });
+        options.push(formattedOption);
       });
+
+      // typescript saying budget is 'never'?
+      const budget = proposal.budget as number;
+
+      const settings = {
+        maxApprovals: proposal.max_options,
+        criteria: proposal.criteria === "Threshold" ? 0 : 1,
+        budgetToken: (parseInt(proposal.budget) > 0
+          ? tenant.contracts.governor.address
+          : ethers.ZeroAddress) as `0x${string}`,
+        criteriaValue:
+          proposal.criteria === "Threshold"
+            ? parseEther(proposal.threshold.toString())
+            : BigInt(proposal.top_choices),
+        budgetAmount: parseEther(budget.toString()),
+      };
+
+      const calldata = encodeAbiParameters(
+        [
+          {
+            name: "proposalOptions",
+            type: "tuple[]",
+            components: [
+              { name: "budgetTokensSpent", type: "uint256" },
+              { name: "targets", type: "address[]" },
+              { name: "values", type: "uint256[]" },
+              { name: "calldatas", type: "bytes[]" },
+              { name: "description", type: "string" },
+            ],
+          },
+          {
+            name: "proposalSettings",
+            type: "tuple",
+            components: [
+              { name: "maxApprovals", type: "uint8" },
+              { name: "criteria", type: "uint8" },
+              { name: "budgetToken", type: "address" },
+              { name: "criteriaValue", type: "uint128" },
+              { name: "budgetAmount", type: "uint128" },
+            ],
+          },
+        ],
+        [options, settings]
+      );
+
+      // TODO: change this so the module addresses are set via the tenant
+      // moving quickly atm
+      const finalApprovalModuleAddress =
+        tenant.namespace === "cyber"
```

Comment:
let's use `TENANT_NAMESPACES.CYBER` instead of the string so we can easily find usages.